lib/cog_api/fake/server.ex

```diff
@@ -60,9 +59,12 @@ defmodule CogApi.Fake.Server do
     end)
   end
 
-  defp update_by_id(list, id, new_resource) do
+  defp update_by_id(list, id, params) do
     index = Enum.find_index(list, fn resource -> resource.id == id end)
-    List.replace_at(list, index, new_resource)
+    old_resource = Enum.find(list, fn resource -> resource.id == id end)
```

Comment:
You have the `index` above, instead of iterating over this again could you use: `Enum.at`?